src/adagrams.js

```diff
@@ -1,15 +1,214 @@
+  const wordArray = word.split('');
+  const handString = lettersInHand.toString();
+  const validLetters = []
+ 
+  for (const letter of word) {
+    if (lettersInHand.includes(letter)) {
+      const wordOccur = wordArray.filter( word => word === letter).length;
+      const handOccur = lettersInHand.filter( hand => hand === letter).length;
+      validLetters.push(letter)
+      if (wordOccur > handOccur) {
+        return false
+      } 
+    }
+    else {
+      return false
+    }
+  if (validLetters.length === word.length) {
+    return true
+  }
+
+  }
+
+      
 };
 
 export const scoreWord = (word) => {
   // Implement this method for wave 3
+  const SCORECHART = {
+    'A':1,
+    'E':1,
+    'I':1,
+    'O':1,
+    'U':1,
+    'L':1,
+    'N':1,
+    'R':1,
+    'S':1,
+    'T':1,
+    'D':2,
+    'G':2,
+    'B':3,
+    'C':3,
+    'M':3,
+    'P':3,
+    'F':4,
+    'H':4,
+    'V':4,
+    'W':4,
+    'Y':4,
+    'K':5,
+    'J':8,
+    'X':8,
+    'Q':10,
+    'Z':10
+  };
+
+  let sumScore = 0;
+  const words = word.toUpperCase();
+
+  
+  
+  for (const letter of words) {
+    let value = SCORECHART[letter];
+    sumScore += value;
+  }
+  if (words.length >= 7 && words.length <= 10) {
+    sumScore += 8;
+  }
+
+
+return sumScore;
+
 };
 
 export const highestScoreFrom = (words) => {
   // Implement this method for wave 4
+  const wordListDict = {};
+  const maxWordList = [];
+  const wordLenTenList = [];
+  const highestWordList = [];
+  let resultDict = {}
+
+  // calculate the highest score of all the words
+  
+  // calculate the score of each word in word_list
+  for (const word of words) {
+    wordListDict[word] = scoreWord(word); 
+  }
+  // calculate highest score in scoreWord dictionary
+  let scores = Object.values(wordListDict);
+  let highestWordScore = Math.max(...scores);
+  
+  // append all words with the highest score to the highest word dictionary
+  
+  for (const [key, value] of Object.entries(wordListDict)) {
+    if (wordListDict[key] === highestWordScore) {
+      highestWordList.push(key)
+    }
+  }
+  // if there is more than one word with the highest score 
+  // and if one of any of those words are 10 letters use that word
+  if (highestWordList.length > 1) {
+    for (const word of highestWordList) {
+      if (word.length === 10) {
+        wordLenTenList.push(word);
+        resultDict['word'] = wordLenTenList[0];
+        resultDict['score'] = highestWordScore;
+        return resultDict
+      }
+      else {
+        const lens = highestWordList.map(words => words.length)
+        console.log(lens)
+        const shortestWord = Math.min(...lens)
+        for (const item of highestWordList) {
+          if (item.length === shortestWord) {
+            resultDict['word'] = item;
+            resultDict['score'] = highestWordScore;
+          }
+        }
+      }  
+    }
+  } 
+  else {
+    resultDict['word'] = highestWordList[0];
+    resultDict['score'] = highestWordScore;
+  }
```

Comment:
I love the logic here, however the nesting can have a negative impact of the readibility of your code. I would consider looking at if some of the functionality of nested code blocks could be combined.